slackbot/manager.py

```diff
@@ -67,8 +67,14 @@ def get_plugins(self, category, text):
         for matcher in self.commands[category]:
             m = matcher.search(text)
             if m:
+                kwargs = m.groupdict()
+                args = tuple(
+                    m.group(i)
+                    for i in range(1, (m.lastindex or 0) + 1)
+                    if i not in matcher.groupindex.values()
+                )
                 has_matching_plugin = True
-                yield self.commands[category][matcher], to_utf8(m.groups())
+                yield self.commands[category][matcher], to_utf8(args), to_utf8(kwargs)
 
         if not has_matching_plugin:
-            yield None, None
+            yield None, None, None
```

Comment:
The [FakeFakePluginManager](https://github.com/lins05/slackbot/blob/c0dbea6d72520fcd94366cf00b2e02a625cd4548/tests/unit/test_dispatcher.py#L13) also need to be updated to return a 3-tuple.